### DISCUSSION TITLE
problem with shift held down and multiple key presses

### DISCUSSION THREAD

**Discussion by kbilsted** [houmain/keymapper#203] [Q&A]
using the mapping 

```
ShiftLeft{!200ms} !AltLeft >> Backspace
AltRight{ShiftLeft}   >> "dddddd"
```

pressing `alt-gr`+ `shift` it prints 'ddddddddd' as expected. Keep holding `alt-gr` down and re-pressing `shift` sends a backspace. I would have expected a sequence of 'ddddddddddd'

if i use

```
@forward-modifiers Shift
ShiftLeft{!200ms} !AltLeft >> Backspace
AltRight{ShiftLeft}   >> "dddddd"
```

the second time i press `shift` nothing happens

**Comment by ristomatti** [comment]
I'm on mobile now, but have you tried if changing the last line to this makes a difference?

```ini
(AltRight ShiftLeft) >> "dddddd"
```

With that said, many mappings seem to break key repeat. I've come to accept it as a matter of compromise that just has to be made case by case.

**Reply by ristomatti** [reply]
...or if you add a `[stage]` before the last line?

**Comment by houmain** [comment]
> `ShiftLeft{!200ms} !AltLeft >> Backspace`

Is there a reason you put the `!AltLeft` after the timeout check?
Doesn't `!AltRight ShiftLeft{!200ms} >> Backspace` do what you want?
